source/common/common/singleton.h

```diff
@@ -0,0 +1,16 @@
+#pragma once
+
+/**
+ * Singleton pattern.
+ */
+template <class T> class Singleton {
+public:
+  /**
+   * Obtain an instance of the singleton for class T.
+   * @return T& a reference to the singleton for class T.
+   */
+  static T& get() {
```

Comment:
OK, I'll add a `TODO` that one day we might want a `MutableSingleton`, but not today.